Description: drop features depending on packages not in Debian yet
 Disable the cryptoauthlib and trusted-service provider.

 In theory, all crates needed by trusted-service-provider are packaged in
 Debian. However, it needs files from another arm project called
 trusted-services-vendor, which has not been packaged yet.

 Also disable the jwt-svid authenticator.

 We will enable the missing features incrementally once the needed crates enter
 Debian.
Author: Emanuele Rocca <ema@debian.org>
Last-Update: 2023-11-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,7 +33,6 @@
     "pkcs11-provider",
     "tpm-provider",
     "mbed-crypto-provider",
-    "cryptoauthlib-provider",
     "direct-authenticator",
 ]
 
@@ -86,7 +85,7 @@
 version = "0.29.1"
 
 [dependencies.picky-asn1]
-version = "0.11.0"
+version = "0.7.0"
 optional = true
 
 [dependencies.picky-asn1-der]
@@ -94,7 +93,7 @@
 optional = true
 
 [dependencies.picky-asn1-x509]
-version = "0.9.0"
+version = "0.10.0"
 optional = true
 
 [dependencies.prost]
@@ -118,10 +117,6 @@
 version = "0.29.0"
 #features = ["bundled"]
 
-[dependencies.rust-cryptoauthlib]
-version = "0.4.5"
-optional = true
-
 [dependencies.sd-notify]
 version = "0.4.1"
 
@@ -132,10 +127,6 @@
 [dependencies.signal-hook]
 version = "0.3.4"
 
-[dependencies.spiffe]
-version = "0.2.1"
-optional = true
-
 [dependencies.threadpool]
 version = "1.8.1"
 
@@ -154,10 +145,6 @@
 version = "0.8.3"
 features = ["small_rng"]
 
-[dev-dependencies.rust-cryptoauthlib]
-version = "0.4.4"
-features = ["software-backend"]
-
 [build-dependencies.bindgen]
 version = "0.72"
 optional = true
@@ -170,18 +157,14 @@
 all-authenticators = [
     "direct-authenticator",
     "unix-peer-credentials-authenticator",
-    "jwt-svid-authenticator",
 ]
 all-providers = [
     "tpm-provider",
     "pkcs11-provider",
     "mbed-crypto-provider",
-    "trusted-service-provider",
 ]
-cryptoauthlib-provider = ["rust-cryptoauthlib"]
-default = ["unix-peer-credentials-authenticator"]
+default = ["unix-peer-credentials-authenticator", "mbed-crypto-provider", "tpm-provider", "pkcs11-provider"]
 direct-authenticator = []
-jwt-svid-authenticator = ["spiffe"]
 mbed-crypto-provider = ["psa-crypto"]
 pkcs11-provider = [
     "cryptoki",
@@ -199,10 +182,4 @@
     "picky-asn1-x509",
     "hex",
 ]
-trusted-service-provider = [
-    "psa-crypto",
-    "bindgen",
-    "prost-build",
-    "prost",
-]
 unix-peer-credentials-authenticator = []
